src/blocking-elements.ts

```diff
@@ -0,0 +1,404 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Symbols for private properties */
+const _blockingElements = Symbol();
+const _alreadyInertElements = Symbol();
+const _topElParents = Symbol();
+const _siblingsToRestore = Symbol();
+const _parentMO = Symbol();
+
+/* Symbols for private static methods */
+const _topChanged = Symbol();
+const _swapInertedSibling = Symbol();
+const _inertSiblings = Symbol();
+const _restoreInertedSiblings = Symbol();
+const _getParents = Symbol();
+const _getDistributedChildren = Symbol();
+const _isInertable = Symbol();
+const _handleMutations = Symbol();
+
+interface Inertable extends HTMLElement {
+  inert?: boolean;
+}
+
+interface InternalState {
```

Comment:
Throwing this here: the BlockingElements class could as well keep track of all this via couple WeakMaps, but that would be quite some refactoring (which probably does belong to a separate PR)